### PR TITLE
[MIR] Implement Mangara's Equity

### PR DESCRIPTION
**Pull request by Cguy7777**

### CODE REVIEW

File: Mage.Sets/src/mage/cards/m/MangarasEquity.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+        game.informPlayers(permanent.getLogName() + ": " + controller.getLogName() + " has chosen " + choice.getChoice());
+        game.getState().setValue(permanent.getId() + "_color", choice.getColor());
+        permanent.addInfo("chosen color", CardUtil.addToolTipMarkTags("Chosen color: " + choice.getChoice()), game);
+
+        return true;
+    }
+
+    @Override
+    public Effect copy() {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/m/MangarasEquity.java:96]: covariant return: this should be `public MangarasEquityColorChoiceEffect copy()`

File: Mage.Sets/src/mage/cards/m/MangarasEquity.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                }
+            }
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public TriggeredAbility copy() {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/m/MangarasEquity.java:152]: covariant return

File: Mage.Sets/src/mage/cards/m/MangarasEquity.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+            if (damagedPermanent == null || !damagedPermanent.isControlledBy(getControllerId())
+                    || !damagedPermanent.isCreature() || !damagedPermanent.getColor().shares(ObjectColor.WHITE)) {
+                return false;
+            }
+        }
+
+        Permanent damageSource = game.getPermanent(event.getSourceId());
+        ObjectColor chosenColor = (ObjectColor) game.getState().getValue(mangarasEquity.getId() + "_color");
+        if (damageSource != null && damageSource.isCreature() && damageSource.getColor().shares(chosenColor)) {
+            for (Effect effect : getEffects()) {
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/m/MangarasEquity.java:139]: why not store `this.getEffects().setValue("damage", damageAmount);` and use `SavedDamageValue` in the constructor of the `DamageTargetEffect`?

File: Mage.Sets/src/mage/cards/m/MangarasEquity.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        } else {
+            Permanent damagedPermanent = game.getPermanent(event.getTargetId());
+            if (damagedPermanent == null || !damagedPermanent.isControlledBy(getControllerId())
+                    || !damagedPermanent.isCreature() || !damagedPermanent.getColor().shares(ObjectColor.WHITE)) {
+                return false;
+            }
+        }
+
+        Permanent damageSource = game.getPermanent(event.getSourceId());
+        ObjectColor chosenColor = (ObjectColor) game.getState().getValue(mangarasEquity.getId() + "_color");
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/m/MangarasEquity.java:137]: miss null check (could be added to `ObjectColor::shares` instead I suppose)

**Comment by Cguy7777** [comment]
My bad on a few spots in there, I should've checked a bit more closely.

File: Mage.Sets/src/mage/cards/m/MangarasEquity.java (hunk 3 above)
Comment (Cguy7777) [reply Mage.Sets/src/mage/cards/m/MangarasEquity.java:139]: Thanks for the tip!

**Pull request merged**